### PR TITLE
8292016: Rework JLI_ReportErrorMessageSys

### PR DESCRIPTION
**Pull request by TheShermanTanker**
JLI_ReportErrorMessageSys has a number of issues, as listed in the linked issue, since it's not widely used, it's easier to remove it and replace the areas it's called in with simpler calls to other error reporting functions:

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8292016](https://bugs.openjdk.org/browse/JDK-8292016): Rework JLI_ReportErrorMessageSys

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9793/head:pull/9793` \
`$ git checkout pull/9793`

Update a local copy of the PR: \
`$ git checkout pull/9793` \
`$ git pull https://git.openjdk.org/jdk pull/9793/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9793`

View PR using the GUI difftool: \
`$ git pr show -t 9793`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9793.diff">https://git.openjdk.org/jdk/pull/9793.diff</a>

</details>

### CODE REVIEW

File: src/java.base/unix/native/libjli/java_md_common.c [hunk 1]
```diff
@@ -204,8 +204,10 @@ JLI_ReportErrorMessageSys(const char* fmt, ...) {
     va_start(vl, fmt);
     vfprintf(stderr, fmt, vl);
 
-    /* TODO: it's safer to use strerror_r, but it isn't available on
-       Solaris 8. Until then.... */
+    /*
+     * TODO: its safer to use strerror_r but is not available on
+     * Solaris 8. Until then....
```
Comment (dholmes-ora) [inline src/java.base/unix/native/libjli/java_md_common.c:209]: We are well past then :) so this should be changed to use strerror_r - though see ./java.base/unix/native/libjava/jni_util_md.c

Comment (TheShermanTanker) [reply src/java.base/unix/native/libjli/java_md_common.c]: Looks like the names in jni_util_md.c are also hopelessly inaccurate - what would equate to strlen(strerror_r(errno)) and returns a size_t is named getLastErrorString :P

**Comment by TheShermanTanker** [comment]
Force push due to some accidental breaking changes

File: src/java.base/unix/native/libjli/java_md_common.c (hunk 1 above)
Comment (TheShermanTanker) [reply src/java.base/unix/native/libjli/java_md_common.c]: Using getLastErrorString in the implementation for Unix, the Windows version of getLastErrorString is also broken, just like JLI_ErrorMessageSys, but unlike the latter shared code does use it, so no attempts will be made to fix it

**Comment by tstuefe** [comment]
Hi Julien,

Just a tip: as long as you are in the active development phase of a patch, you can mark the PR as "draft". That way you do not spam the mailing lists with updates for every single push. And you save valuable Reviewer cycles since they won't have to look at the half-finished review.

Once you are content with the patch, and all tests are green, you can mark the PR as "ready for review". Only then Review mails are sent out.

---

That said, I don't think printing both codes is an improvement. See my comment in JBS.

Cheers, Thomas

**Comment by TheShermanTanker** [comment]
Alright Skara, I'm really sorry but don't lynch me for what I'm about to do...

**Comment by TheShermanTanker** [comment]
Closing in favour of a newer rewrite

**Pull request closed without merging**